src/dc_mimefactory.rs

```diff
@@ -958,12 +954,14 @@ pub unsafe fn dc_mimefactory_render(
             let human_mime_part: *mut mailmime = build_body_text(message_text);
             mailmime_add_part(multipart, human_mime_part);
             let version = dc_get_version_str();
+            let from_addr_c = CString::yolo(factory.from_addr.as_ref().unwrap().as_str());
+            let from_addr_ptr = from_addr_c.as_ptr();
             message_text2 =
                 dc_mprintf(
                     b"Reporting-UA: Delta Chat %s\r\nOriginal-Recipient: rfc822;%s\r\nFinal-Recipient: rfc822;%s\r\nOriginal-Message-ID: <%s>\r\nDisposition: manual-action/MDN-sent-automatically; displayed\r\n\x00"
                         as *const u8 as *const libc::c_char,
                     version,
-                    factory.from_addr, factory.from_addr,
+                    from_addr_ptr, from_addr_ptr,
                     factory.msg.rfc724_mid
```

Comment:
Again, `unwrap` is justified by the fact, that `%s` formatter does not allow NULL, so semantic is not changed.